src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -0,0 +1,705 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Security;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Data.SqlClient
+{
+    sealed internal class LastIOTimer
+    {
+        internal long _value;
+    }
+
+    partial class TdsParserStateObject
+    {
+        private static int _objectTypeCount; // EventSource counter
```

Comment:
Yep, we squash and merge. So, feel free to prepare separate commit(s). 🙂

I've already addressed the conflict with the baseline. So, rebasing is not required here.

Reply:
> I've already addressed the conflict with the baseline. So, rebasing is not required here.

Just to be clear, I was talking about rebasing the branch in order to edit its commits (to avoid introducing a separate commit for the fixes), but it doesn't matter since they will be squashed afterwards.

@Wraith2 I used to avoid rebasing like the plague for the same reason you do, but after learning how to use it properly (and with the help of a UI), it has become my favorite git tool. By the way, git has another nice feature, the reflog, which basically allows you to bring your repository back to an earlier state, in case you have messed up (e.g. to undo a rebase that went wrong).